lib/redis/ick.rb

```diff
@@ -301,8 +286,113 @@ def ickcommit(ick_key,*members)
       end
       _statsd_increment('profile.ick.ickcommit.calls')
       _statsd_timing('profile.ick.ickcommit.members',members.size)
+      raw_results = nil
       _statsd_time('profile.ick.time.ickcommit') do
-        _eval(LUA_ICKCOMMIT,ick_key,*members)
+        raw_results = _eval(
+          LUA_ICKEXCHANGE,
+          ick_key,
+          0,
+          false,              # backwash not relevant in ickcommit
+          *members
+        )
+      end
+      if raw_results.is_a?(Redis::Future)
+        class << raw_results  # extend the future with our own continuation
```

Comment:
So every time you get a Redis::Future as a result, you will run the code that monkey-patches the Redis::Future class?

Reply:
OK I have some comments on two levels:

More severely, it feels weird to need to post-process the results of a Lua script. Could the Lua script not 'just' return the result that the ruby code expects?

Less severely, if there is post-processing to be done on a result, maybe it would make sense to capture the pattern to avoid code duplication of the post-processing pattern, and the post-processing code itself?